### ISSUE TITLE
Getting started with Windows development

### ISSUE THREAD

**Issue by richb-hanover** [PRQL/prql#3132]
### What's up?

@max-sixty @not-my-profile I see from #3119 that Windows development still needs testing. I recently came into a Windows laptop that I use irregularly. It's modestly powerful - i7, 16GB RAM, SSD. It runs the Dev Container fine, but I don't know how to get started using native Windows development tools

What do I need? I know I'll need to install VS Code, compilers, and Task (there are a zillion tutorials on the web to help with that). What's a good shell/command line program? And do I then just type `task setup-dev` and see where the smoke comes out? Thanks.

**Comment by eitsupi** [comment]
I tried to build on Windows in #2894, but could not get duckdb-rs to build successfully after all.

Thus the short answer is "Use WSL2".

**Comment by richb-hanover** [comment]
> Thus the short answer is "Use WSL2".

That info should go into the new Project section of the new documentation (unless it already is...)

**Comment by eitsupi** [comment]
> That info should go into the new Project section of the new documentation (unless it already is...)

However, this is because I am not used to developing on Windows, and if one is skilled in building environments on Windows, most things should work... (In fact, it works on CI)

**Comment by max-sixty** [comment]
FYI #3132 is only about the GitHub Actions cache.

+1 re using WSL.

Though now that duckdb-rs is optional, it might well be possible to develop on Windows directly. But we'd probably need someone who wanted to own that to confirm, and ensure it stayed that way...

**Comment by richb-hanover** [comment]
Yes. I am completely content to advise people to use WSL. My thoughts:

- We want to welcome people who happen to use Windows computers to contribute. Our documentation should tell them how to do it.
- This project is platform-independent: its output (the PRQL compiler) doesn't matter what tools people use to develop
- It's a very geeky project - anyone with the chops to make a difference won't care whether they're using WSL or some set of native Windows development tools.
- Therefore, I would advocate that we **not** spend much time documenting and maintaining the native Windows development  process, but instead make sure our documentation helps people easily use WSL

This might leave people who desperately want to use native Windows development tools out in the cold, but I suspect that cohort is vanishingly small. (And they can be the ones to own that process...)

**Comment by richb-hanover** [comment]
PS Any tips on using WSL? Thanks.

**Comment by max-sixty** [comment]
Great! I will leave it to others who use windows to own the docs, but agree with those points @richb-hanover !
